### PR TITLE
feat(mobile/background): use reducedUI only when app is not active and log appState

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

**Review by saghul — APPROVED** [review]

**Review by saghul — APPROVED** [review]

File: react/features/base/lastn/middleware.ts [hunk 1]
```diff
@@ -44,7 +44,11 @@ const _updateLastN = debounce(({ dispatch, getState }: IStore) => {
     // 3. -1 as the default value.
     let lastNSelected = config.startLastN ?? (config.channelLastN ?? -1);
 
-    if (appState === 'background' || carMode) {
+    // Because this is shared, on web appState is always undefined,
+    // meaning that it is never active
+    const finalAppState = navigator.product === 'ReactNative' ? appState !== 'active' : appState === 'background';
```
Comment (saghul) [inline react/features/base/lastn/middleware.ts:49]: THis reads very weird, if you write it like this it's more obvious:

```
if (navigator.product === 'ReactNative' && (appState !== 'active' || carMode)) {
...
}
```

File: react/features/calendar-sync/middleware.ts [hunk 2]
```diff
@@ -71,6 +71,11 @@ MiddlewareRegistry.register(store => next => action => {
  * @returns {void}
  */
 function _maybeClearAccessStatus(store: IStore, { appState }: { appState: string; }) {
-    appState === 'background'
+
+    // Because this is shared, on web appState is always undefined,
+    // meaning that it is never active
+    const finalAppState = navigator.product === 'ReactNative' ? appState !== 'active' : appState === 'background';
```
Comment (saghul) [inline react/features/calendar-sync/middleware.ts:77]: Ditto.

File: react/features/mobile/call-integration/middleware.ts [hunk 3]
```diff
@@ -272,10 +277,14 @@ function _conferenceWillJoin({ dispatch, getState }: IStore, next: Function, act
         return result;
     }
 
-    // When assigning the call UUID, do so in upper case, since iOS will return
+    // When assigning the callUUID, do so in upper case, since iOS will return
     // it upper-cased.
     conference.callUUID = (callUUID || uuidv4()).toUpperCase();
 
+    if (callUUID) {
```
Comment (saghul) [inline react/features/mobile/call-integration/middleware.ts:284]: You should be able to drop this now. The reason why we didn't find the call earlier is because the CallKit provider was reset.

File: react/features/mobile/call-integration/middleware.ts [hunk 4]
```diff
@@ -362,12 +372,18 @@ function _onPerformEndCallAction({ callUUID }: { callUUID: string; }) {
...
+    if (conference?.callUUID === callUUID) {
         // We arrive here when a call is ended by the system, for example, when
         // another incoming call is received and the user selects "End &
         // Accept".
+
         delete conference.callUUID;
+
         dispatch(appNavigate(undefined));
+
+        logger.warn('A new incoming call has been accepted!');
```
Comment (saghul) [inline react/features/mobile/call-integration/middleware.ts:386]: Drop the bogus log line

File: react/features/calendar-sync/middleware.ts [hunk 5]
```diff
@@ -71,6 +71,9 @@ MiddlewareRegistry.register(store => next => action => {
  * @returns {void}
  */
 function _maybeClearAccessStatus(store: IStore, { appState }: { appState: string; }) {
-    appState === 'background'
-        && store.dispatch(setCalendarAuthorization(undefined));
+
+    // Because this is shared, on web appState is always undefined,
+    // meaning that it is never active
+    return navigator.product === 'ReactNative' && (appState !== 'active'
```
Comment (saghul) [inline react/features/calendar-sync/middleware.ts:77]: I think in this case we want the previous behavior. There is no point in resetting the permission so often.

File: react/features/mobile/call-integration/middleware.ts (hunk 3 above)
Comment (Calinteodor) [reply react/features/mobile/call-integration/middleware.ts]: Addressed.

File: react/features/mobile/call-integration/middleware.ts (hunk 4 above)
Comment (Calinteodor) [reply react/features/mobile/call-integration/middleware.ts]: Dropped.

File: react/features/calendar-sync/middleware.ts [hunk 6]
```diff
@@ -71,6 +71,5 @@ MiddlewareRegistry.register(store => next => action => {
  * @returns {void}
  */
 function _maybeClearAccessStatus(store: IStore, { appState }: { appState: string; }) {
-    appState === 'background'
-        && store.dispatch(setCalendarAuthorization(undefined));
+    return appState === 'background' && store.dispatch(setCalendarAuthorization(undefined));
```
Comment (saghul) [inline react/features/calendar-sync/middleware.ts:74]: LEt's undo this change pl

File: react/features/mobile/call-integration/middleware.ts [hunk 7]
```diff
@@ -362,11 +366,14 @@ function _onPerformEndCallAction({ callUUID }: { callUUID: string; }) {
     const { dispatch, getState } = this; // eslint-disable-line @typescript-eslint/no-invalid-this
     const conference = getCurrentConference(getState);
 
-    if (conference && conference.callUUID === callUUID) {
+    logger.warn('Handling Callkit performEndCallAction event');
```
Comment (saghul) [inline react/features/mobile/call-integration/middleware.ts:369]: Let's drop this pl

File: react/features/mobile/call-integration/middleware.ts [hunk 8]
```diff
@@ -383,10 +390,14 @@ function _onPerformSetMutedCallAction({ callUUID, muted }: { callUUID: string; m
     const { dispatch, getState } = this; // eslint-disable-line @typescript-eslint/no-invalid-this
     const conference = getCurrentConference(getState);
 
-    if (conference && conference.callUUID === callUUID) {
+    logger.warn('Handling Callkit performSetMutedCallAction event');
```
Comment (saghul) [inline react/features/mobile/call-integration/middleware.ts:393]: Let's drop the log line pl

**Review by saghul — APPROVED** [review]
